webapp/src/components/add_issue/add_issue.jsx

```diff
@@ -0,0 +1,328 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+    makeStyleFromTheme,
+    changeOpacity,
+} from 'mattermost-redux/utils/theme_utils';
+
+import TextareaAutosize from 'react-textarea-autosize';
+
+import FullScreenModal from '../modals/modals.jsx';
+import Button from '../../widget/buttons/button';
+import Chip from '../../widget/chip/chip';
+import AutocompleteSelector from '../user_selector/autocomplete_selector.tsx';
+import './add_issue.scss';
+import CompassIcon from '../icons/compassIcons';
+import {getProfilePicture} from '../../utils';
+
+const PostUtils = window.PostUtils;
+
+export default class AddIssue extends React.PureComponent {
+    static propTypes = {
+        visible: PropTypes.bool.isRequired,
+        message: PropTypes.string.isRequired,
+        postID: PropTypes.string.isRequired,
+        assignee: PropTypes.object,
+        closeAddBox: PropTypes.func.isRequired,
+        submit: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        autocompleteUsers: PropTypes.func.isRequired,
+        openAssigneeModal: PropTypes.func.isRequired,
+        removeAssignee: PropTypes.func.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            message: '',
+            description: '',
+            sendTo: null,
+            attachToThread: false,
+            previewMarkdown: false,
+            assigneeModal: false,
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.visible && !state.message && props.message !== state.message) {
+            return {message: props.message};
+        }
```

Comment:
I'm wondering if this would cause an issue with:

- I'm editing the todo message
- I remove all text
- I do _some_ action that causes `getDerivedStateFromProps` to run (some prop or state value changes like `description`)
- The message has been changed from empty to `props.message`

Why not set this in the constructor instead?

Reply:
@asaadmahmood Testing the PR out, this bug is present in the new code. Repro steps:

- From the post actions menu, click `Add Todo`
- Note that the todo message is pre-populated
- Erase all of the text of the message

Expected: The message field is empty

Observed: The message field's contents reverts to the original value of the post's message